src/movies.js

```diff
@@ -1,25 +1,192 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+/*
+{
+  title: 'The Shawshank Redemption',
+  year: 1994,
+  director: 'Frank Darabont',
+  duration: '2h 22min',
+  genre: ['Crime', 'Drama'],
+  score: 9.3
+},
+*/
+function getAllDirectors(moviesArray) {
+  return moviesArray.map(el => el.director);
+
+  /*****  returning the array filtered. ******
+  return moviesArray
+    .map(el => el.director)
+    .filter((director, index, self) => self.indexOf(director) === index);
+  */
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  return moviesArray.filter(el => el.director === 'Steven Spielberg' && el.genre.includes('Drama')).length;
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  let len = moviesArray.length;
```

Comment:
Si no se va a reasignar el valor de len en ningun sitio mejor usar const!
```suggestion
  const len = moviesArray.length;
```